### DISCUSSION TITLE
How to configure the downstream protocol and protocol version?

### DISCUSSION THREAD

**Discussion by 842549829** [ThreeMammals/Ocelot#1867] [Q&A] [answered]
Hi,
About the downstream protocol version (http2 vs http3)...

The current http protocol of my downstream service is http2.0 and http3.0.
How to configure the downstream protocol and protocol version?

**Comment by raman-m** [comment]
Hi [liuxiaoji](https://github.com/842549829)!
You can ensure about current supported HTTP versions in this paragraph: [DownstreamHttpVersion](https://ocelot.readthedocs.io/en/latest/features/configuration.html?highlight=DownstreamHttpVersion#downstreamhttpversion)
I believe Ocelot doesn't support HTTP 3. It requires testing and development with usage of new `HttpClient` class which support HTTP 3.
And, the **DownstreamHttpVersion** property is a part of [Route](https://ocelot.readthedocs.io/en/latest/features/routing.html).

You can try to specify HTTP3 route and let us know your testing results, please.

P.S. MS Docs: [HTTP/3 Server](https://learn.microsoft.com/en-us/dotnet/core/extensions/httpclient-http3#http3-server)

**Reply by raman-m** [reply, marked as answer]
Ocelot should be able to support both Http2 and Http3.
To use appropriate protocol version use my advice above ☝️ 

Also, please watch this PR #1673 which is related to default version policy
- #1673 
- #1672 

Hope it helps!

**Comment by 842549829** [comment]
thank you.

**Reply by raman-m** [reply]
Welcome back with PRs!
